src/main/java/com/klarna/hiverunner/HiveRunnerExtension.java

```diff
@@ -51,7 +51,7 @@
   private final HiveRunnerConfig config = new HiveRunnerConfig();
   private Path basedir;
   private HiveShellContainer container;
-  protected List<Script> scriptsUnderTest = new ArrayList<Script>();
+  protected List<Script> scriptsUnderTest = null;
```

Comment:
One of the reasons to have an empty list instead of null is to avoid having NullPointerExceptions and null checks. For example, if you have a for loop over the list and the list is empty it won't do anything and you don't need a null check. Where is this check being done? Could it be changed to check if the list is null or empty?

Reply:
The check is being done [here](https://github.com/klarna/HiveRunner/blob/a0310a9bbc183bd2d182e1bff5a84b6025632ce9/src/main/java/com/klarna/hiverunner/HiveRunnerCore.java#L67), which is what was causing the problem. When using the extension, this `if` statement was always true, whereas in the runner it wasn't always true. So I think the problem was actually that the null check for the scripts wasn't doing anything in the extension.

Reply:
Yes, what I meant was, can we keep the empty list here and change the check in that other place? Perhaps also change the runner to have an empty list instead of `null`. Then an empty list will be passed in to `HiveShellBuilder` but I think that's OK as that's OK as that's what it does by default - https://github.com/klarna/HiveRunner/blob/a0310a9bbc183bd2d182e1bff5a84b6025632ce9/src/main/java/com/klarna/hiverunner/builder/HiveShellBuilder.java#L37